Modules/_ssl.c

```diff
@@ -157,6 +157,25 @@ static void _PySSLFixErrno(void) {
 # define HAVE_ALPN
 #endif
 
+/* NPN support defines changed in OpenSSL 1.1
+ * LibreSSL 2.6.1 no longer supports NPN but does not set the
+ * designated OPENSSL_NO_NEXTPROTONEG feature flag. See upstream issue
+ * https://github.com/libressl-portable/portable/issues/368
+ */
+#ifdef OPENSSL_VERSION_1_1
+# ifdef OPENSSL_NO_NEXTPROTONEG
```

Comment:
I think it is safe to simplify the check even further. NPN is available if ``TLSEXT_TYPE_next_proto_neg`` and ``OPENSSL_NO_NEXTPROTONEG`` is not defined.

```
#ifdef OPENSSL_NO_NEXTPROTONEG
#  define HAVE_NPN 0
#elif defined(TLSEXT_TYPE_next_proto_neg)
#  define HAVE_NPN 1
#else
#  define HAVE_NPN 0
# endif
```